HLTrigger/Muon/plugins/HLTMuonDimuonL3Filter.h

```diff
@@ -59,29 +59,31 @@ class HLTMuonDimuonL3Filter : public HLTFilter {
   bool previousCandIsL2_;
-  bool fast_Accept_;                 // flag to save time: stop processing after identification of the first valid pair
-  int min_N_;                        // minimum number of muons to fire the trigger
-  double max_Eta_;                   // Eta cut
-  int min_Nhits_;                    // threshold on number of hits on muon
-  double max_Dr_;                    // impact parameter cut
-  double max_Dz_;                    // dz cut
-  int chargeOpt_;                    // Charge option (0:nothing; +1:same charge, -1:opposite charge)
-  std::vector<double> min_PtPair_;   // minimum Pt for the dimuon system
-  std::vector<double> max_PtPair_;   // miaximum Pt for the dimuon system
-  std::vector<double> min_PtMax_;    // minimum Pt for muon with max Pt in pair
-  std::vector<double> min_PtMin_;    // minimum Pt for muon with min Pt in pair
-  std::vector<double> max_PtMin_;    // maximum Pt for muon with min Pt in pair
-  std::vector<double> min_InvMass_;  // minimum invariant mass of pair
-  std::vector<double> max_InvMass_;  // maximum invariant mass of pair
-  double min_Acop_;                  // minimum acoplanarity
-  double max_Acop_;                  // maximum acoplanarity
-  double min_PtBalance_;             // minimum Pt difference
-  double max_PtBalance_;             // maximum Pt difference
-  double nsigma_Pt_;                 // pt uncertainty margin (in number of sigmas)
-  double max_DCAMuMu_;               // DCA between the two muons
-  double max_YPair_;                 // |rapidity| of pair
-  bool cutCowboys_;                  ///< if true, reject muon-track pairs that bend towards each other
-  const edm::InputTag theL3LinksLabel;                                //Needed to iterL3
+  bool fast_Accept_;  // flag to save time: stop processing after identification of the first valid pair
+  int min_N_;         // minimum number of muons to fire the trigger
+  double max_Eta_;    // Eta cut
+  int min_Nhits_;     // threshold on number of hits on muon
+  double max_Dr_;     // impact parameter cut
+  double max_Dz_;     // dz cut
+  int chargeOpt_;     // Charge option (0:nothing; +1:same charge, -1:opposite charge)
+  const std::vector<double> min_PtPair_;              // minimum Pt for the dimuon system
+  const std::vector<double> max_PtPair_;              // miaximum Pt for the dimuon system
+  const std::vector<double> min_PtMax_;               // minimum Pt for muon with max Pt in pair
+  const std::vector<double> min_PtMin_;               // minimum Pt for muon with min Pt in pair
+  const std::vector<double> max_PtMin_;               // maximum Pt for muon with min Pt in pair
+  const std::vector<double> min_InvMass_;             // minimum invariant mass of pair
+  const std::vector<double> max_InvMass_;             // maximum invariant mass of pair
+  const std::vector<int> invertDiMuonMassSelection_;  // condition to invert the invariant mass selection
```

Comment:
```suggestion
```

Reply:
In principle, I could choose two mass windows, and then I do not need to add the invert mass window selection choice. However, this increases the number of conditions checked from 19 to 28, for every event with a muon pair because now we have to check against a pair of values for each of the following vectors.
  - min_PtMin_
  - min_PtMax_
  - max_PtMin_
  - min_PtPair_
  - max_PtPair_
 
This can potentially cause a significant increase in the timing of an event. Please let me know what are your thoughts on this.

Reply:
> This can potentially cause a significant increase in the timing of an event. Please let me know what are your thoughts on this.

This would have to be quantified (although I would be surprised if this specific filter, even in its possible new configurations, would add significant time to the overall HLT reconstruction). Assuming there will be a new path which uses a new configuration for this plugin (inverted mass-windows), is the timing of that path significant, and is this filter a significant contributor to that? (I'm not asking now, but these are the questions, imho). Readibility of the code, and making sure the default behaviour of the plugin is preserved upon the introduction of new optional features are also important factors.

Reply:
The expected rate of the HLT that this path will be used in is ~5Hz. This filter is the last step in this path. Dimuon triggers typically take less time to run at HLT. Do you think I should make a time study regarding this? Or is there any way we can improve the readability of the code with the added option to invertDiMuonMass and move forward?

Reply:
Based on this info, the easiest solution in my opinion is to not modify the cpp implementation of the plugin (which is already not particularly easy to read..), and to simply configure it differently at the python level. If later on it turns out that there is creates a significant timing increase (before the new path is integrated), then we'll have to think about improving the plugin.

By the way, if `invertDiMuonMass` is not removed, the PR as is will not pass the tests. The menus would have to be customised appropriately in order for the relevant modules to use the correct values of the new `vint` (a list of zeros, I guess, but with different size for different modules). This change is not terribly complex, but it is missing nonetheless at the moment.